### DISCUSSION TITLE
Make Select QRTZ_LOCK timeout configurable

### DISCUSSION THREAD

**Discussion by vwitteki** [quartznet/quartznet#1450] [Ideas] [answered]
Hello!  In the latest quartznet version, the MaxRetry and RetryPeriod for getting the TRIGGER_ACCESS lock are both configurable (defaults 3 tries and 1000ms, respectively).
However, it seems like the command timeout per SQL call is not configurable (default 30s https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlcommand.commandtimeout?view=dotnet-plat-ext-6.0#property-value).
So even if we configure MaxRetry to 0 and RetryPeriod to 0, we will still be waiting up to 30s for the command to timeout.

Ultimately, the behavior we would like to see is for our application to 'fail faster' if we can't get the lock.  So we would like the CommandTimeout to be configurable.

Thoughts? Is there a configuration I missed, or a workaround to effect this kind of behavior?
If not, could we make a PR for this?

**Comment by lahma** [comment, marked as answer]
Sounds that this is missing. I guess some providers allow this via connection string but maybe not MS SQL Server. If it's "settable" in all providers I guess we can add is as an configuration option, otherwise needs to be injected to specific DB provider.

A PR would be great, thank you.

**Reply by vwitteki** [reply]
Sounds good, thank you.
FYI only- for SQL Server I think you can set the _connection_ timeout in the connection string, but not the timeout for commands executed over that connection.
